docs/tech-specs/pull-consumer-spec.md

```diff
@@ -0,0 +1,42 @@
+# Tech Spec: Adding Pull Consumer Support
+
+<!-- |  |  |
+| -- | -- |
+| *Version* | 1 |
+| *By* | Imran M Yousuf |
+| *Date* | November 10, 2020 |
+| *Last Update* | December 30, 2020 | -->
+
+## Problem Statement / Goal
```

Comment:
In addition to what you said, you can also add that this is way for the consumer to throttle/rate-limit consumption of events.